src/Controller/EndpointsController.php

```diff
@@ -90,71 +91,83 @@ public function addonCancel() {
     // Badge not found, but since this addon was cancelled it's inconsequential
     Log::error('WHMCS addon cancellation successfully removed a family member user badge in Maker Manager for user with WHMCS user id ' . $this->request->data['userid'] . ', service id ' . $this->request->data['serviceid'] . ' and addon id ' . $this->request->data['addonid'], ['scope' => ['users']]);
   }
+
+
     public function userAdd(){
-        $usersTable = TableRegistry::get('Users');
-        if($existing_user = $usersTable->find()->where(['whmcs_user_id' => $this->request->data['user_id']])->first())
-        {
-            if (!empty($existing_user)) {
-                // Create a new active directory account, if one didn't previously exist
-                $chars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
-                $password = substr(str_shuffle($chars), 0, 16);
-                $existing_user->createActiveDirectoryAccount($password, true);
-
-                $existing_user->changeActiveDirectoryPassword($this->request->data['password']);
-
-                Log::error('WHMCS client successfully changed Active Directory password for WHMCS user id ' . $this->request->data['userid'] . '.', ['scope' => ['users']]);
-            } else {
-                $this->_sendEmail('WHMCS client failed Active Directory password changing in Maker Manager for WHMCS user id ' . $this->request->data['user_id'] . '. WHMCS user not found in Maker Manager.');
-                Log::error('WHMCS client failed Active Directory password changing in Maker Manager for WHMCS user id ' . $this->request->data['user_id'] . '. WHMCS user not found in Maker Manager.', ['scope' => ['users']]);
-            }
-        }
+      Log::error('userAdd: called '.microtime());
+      Log::error('Data passed: '.var_export($this->request->data, true));
```

Comment:
Using `Log:error()` for debugging goes against best practices, as an error should only be logged when we encounter an error. Additionally, logging data such as an entire request body (`$this->request->data`) should only be done when in debug mode as it can and will contain potentially sensitive info.

Reply:
Nitpick: indentation is different here (2 spaces) than other methods (4 spaces). [PSR-12](https://www.php-fig.org/psr/psr-12/#24-indenting) recommends 4 spaces per indent level.

Reply:
That's a good question. It was already present for creating the active directory user (see lines 162-164) in the original file.

I just moved it and shortened it up a bit.

Reply:
Agreed, these should have been pulled. 

I'll modify this.